hyperapp.d.ts

```diff
@@ -55,40 +55,44 @@ export function h<Props>(
 /** @namespace [App] */
 
 /** The application state.
- * 
+ *
  * @memberOf [App]
  */
 export interface State {}
 
+export interface Update<State extends Hyperapp.State> {
+  (value: Partial<State>): void
```

Comment:
How would it work with the new namespace feature ? `Update<State[namespace]>` ?

Reply:
I'm not sure. That's actually one of things that makes me nervous about hyperapp.

Reply:
Actually I think it just works: in each module you define `interface ModuleState` and `interface ModuleActions extends Hyperapp.Actions<ModuleState>` and in each of the actions, you may return a `Thunk<ModuleState>`. There is an example of this (without thunk) in https://github.com/hyperapp/hyperapp/blob/master/test/ts/index.tsx.